fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/random/RegexGenerator.java

```diff
@@ -49,6 +63,38 @@ public final class RegexGenerator {
 		PREDEFINED_CHARACTER_CLASSES = Collections.unmodifiableMap(characterClasses);
 	}
 
+	public String generate(String regex, int[] flags, @Nullable Integer min, @Nullable Integer max, boolean notBlank) {
+		boolean caseSensitive = Arrays.stream(flags).noneMatch(it -> it == FLAG_CASE_INSENSITIVE);
+
+		try {
+			RgxGen rgxGen = generateRgxGen(regex, caseSensitive);
+
+			int minLength = min == null ? 0 : min;
+			int maxLength = max == null ? Integer.MAX_VALUE : max;
+
+			String result = executor.submit(() ->
+				rgxGen.stream()
+					.filter(it -> it.length() >= minLength && it.length() <= maxLength)
+					.filter(it -> !(notBlank && it.trim().isEmpty()))
+					.limit(DEFAULT_REGEXP_GENERATION_MAX_SIZE)
+					.findFirst()
+			).get(DEFAULT_REGEXP_GENERATION_TIMEOUT_SEC, TimeUnit.SECONDS).orElse(null);
+
+			checkValidity(regex, result, caseSensitive);
+			return result;
+		} catch (Exception ex) {
+			throw new IllegalArgumentException(
+				String.format(
+					"String generation failed for the regular expression \"%s\" provided in @Pattern."
```

Comment:
You don't need to specify that you used `@Pattern`, `RegexGenerator ` is not just for `@Pattern`